### ISSUE TITLE
Opening up Templating/RazorTemplateBase.cs

### ISSUE THREAD

**Issue by bchavez** [RazorGenerator/RazorGenerator#36]
I was wondering if we could open up [`RazorTemplateBase.cs`](https://github.com/RazorGenerator/RazorGenerator/blob/master/RazorGenerator.Templating/RazorTemplateBase.cs) with `protected` *fields* and `virtual` *methods*. I think it would be good since it would allow a lot of customization. In particular, I'd like to implement a custom `@section` behavior.

At the moment, i'm struggling to make it happen because some necessary method calls and variables are `private`/non-`virtual`.

Would you accept a pull request if I made the changes?

Thanks,
Brian

**Comment by davidebbo** [comment]
Yes, I think we can do that. Maybe not everything, but start with what you need for extensibility. Note that you also have the option of not using that class at all and provide a replacement that just has methods by the same name (sort of duck typing).

**Comment by bchavez** [comment]
Hi @davidebbo , just wanted to say thank you again for all your work with RazorGenerator. I opted for duck typing / replacing `RazorTemplateBase` implementation for maximum rendering control. It seemed like the best option for [my use case here](https://github.com/bchavez/RethinkDb.Driver/blob/master/Source/Templates/CodeGen/RazorTemplateBase.cs). I needed to dig deep into the internals to get `@section` to work the way I wanted it to (using `@section` like a block replacement template).

Your implementation is simple enough (which is really great) that I don't mind maintaining this local copy. I'm thinking now it's probably best to leave your implementation alone as it currently stands.
